reactor-core/src/test/java/reactor/core/publisher/FluxSwitchOnFirstTest.java

```diff
@@ -98,6 +101,71 @@ public void shouldNotSubscribeTwiceConditional() {
                   .hasMessage("FluxSwitchOnFirst allows only one Subscriber");
     }
 
+    @Test
+    public void shouldNotSubscribeTwiceWhenCanceled() {
+        Schedulers.single().start();
+        CountDownLatch latch = new CountDownLatch(1);
+        StepVerifier.create(Flux.just(1L)
+                                .doOnComplete(() -> {
+                                    try {
+                                        latch.await();
+                                    }
+                                    catch (InterruptedException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                .hide()
+                                .publishOn(Schedulers.parallel())
+                                .cancelOn(NoOpsScheduler.INSTANCE)
+                                .doOnCancel(latch::countDown)
+                                .switchOnFirst((s, f) -> f)
+                                .doOnSubscribe(s ->
+                                    Schedulers
+                                        .single()
+                                        .schedulePeriodically(s::cancel, 10, 0, TimeUnit.MILLISECONDS)
```

Comment:
Why not [Scheduler#schedule(Runnable task, long delay, TimeUnit unit)](https://projectreactor.io/docs/core/release/api/reactor/core/scheduler/Scheduler.html#schedule-java.lang.Runnable-long-java.util.concurrent.TimeUnit-)?

Reply:
fixed